File: solucoes/2021/nivel jr/fase 3/py/sapo.py
M, N = map(int, input().split())
P = int(input())
lago = []
is_possible = 'S'
for i in range(P):
    posicoes = list(map(int, input().split()))
    lago.append(posicoes)

for i in range(P-1):
    cond_X = lago[i][0] - lago[i+1][0] > 3 or lago[i+1][0] - lago[i][0] > 3
    cond_Y = lago[i][1] - lago[i+1][1] > 3 or lago[i+1][1] - lago[i][1] > 3 
    if(cond_Y or cond_X):
        is_possible = 'N'
        break  

S_C, S_L = map(int, input().split())
R_C, R_L = map(int, input().split())

print(is_possible)

File: solucoes/2021/nivel jr/fase 3/js/sapo.js
var M, N, P, L, C, S_C, S_L, R_C, R_L 

scanf("%d%d%d", "M", "N", "P")

var lago = []
var is_possible = 'S'

for(var i = 0; i < P; i++){
    var Col = []
    scanf("%d%d", "L", "C")
    Col.push(L)
    Col.push(C)
    lago.push(Col)
}

for(var i = 0; i < P-1; i++){
    var cond_X = lago[i][0] - lago[i+1][0] > 3 || lago[i+1][0] - lago[i][0] > 3
    var cond_Y = lago[i][1] - lago[i+1][1] > 3 || lago[i+1][1] - lago[i][1] > 3 
    if(cond_Y || cond_X){
        is_possible = 'N'
        break  
    }
}

scanf("%d%d", "S_C", "S_L")
scanf("%d%d", "R_C", "R_L")

printf("%s", is_possible)